### PR TITLE
KAFKA-2379: Add basic documentation for Kafka Connect.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@gwenshap @junrao Not sure who wants to review/commit. User guide needs a lot more detail, but the very basics are there. The dev guide is more complete, but could use a sanity check since it's an updated version from ages ago. Would like to get more in there around design as well, but that's a much more substantial project.

File: docs/quickstart.html [hunk 1]
```diff
@@ -187,3 +187,67 @@
...
+
+The connectors continue to process data, so we can add data to the file and see it move through the pipeline:
+
+<pre>
+&gt; <b>echo "Another line" >> test.txt</b>
+</pre>
+
+You should see the line appear in the console consumer output and in the sink file.
+
+This
```
Comment (gwenshap) [inline docs/quickstart.html]: Last line seems odd...

**Comment by gwenshap** [comment]
Aren't we missing documentation (or auto-generation) of the configuration properties?

File: docs/quickstart.html (hunk 1 above)
Comment (ewencp) [inline docs/quickstart.html]: Ooops, I must have thought I had more to say, but apparently I did not :) removed.

**Comment by ewencp** [comment]
@gwenshap Added the autogenerated docs. We may need to adjust the important levels for some of them -- I think they fell out of date as we refined how we managed internal serialization, the converters vs serializers, etc.

My gradle-fu is not strong, you should probably double check the way the dependencies are setup works out ok -- these are the first auto-generated docs where the class doesn't live in core.

Also, not entirely sure about the placement. It's a bit awkward to fit Kafka Connect into the docs with the current structure.

**Comment by gwenshap** [comment]
LGTM. Thank you for the docs!

**Comment by gwenshap** [comment]
I think the placement is good.

**Comment by gwenshap** [comment]
Codegen looks good too.

You are right on mis-ordering, "internal.key.converter" appears way too high on the list :)
We can fix this later.

**Pull request closed without merging**